homeassistant/components/system_bridge/coordinator.py

```diff
@@ -0,0 +1,137 @@
+"""DataUpdateCoordinator for System Bridge."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+from typing import Callable
+
+from systembridge import Bridge
+from systembridge.client import BridgeClient
+from systembridge.exceptions import (
+    BridgeAuthenticationException,
+    BridgeConnectionClosedException,
+    BridgeException,
+)
+from systembridge.objects.events import Event
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_PORT,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import BRIDGE_CONNECTION_ERRORS, DOMAIN
+
+
+class SystemBridgeDataUpdateCoordinator(DataUpdateCoordinator[Bridge]):
+    """Class to manage fetching System Bridge data from single endpoint."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        LOGGER: logging.Logger,
+        *,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize global System Bridge data updater."""
+        self.bridge = Bridge(
+            BridgeClient(async_get_clientsession(hass)),
+            f"http://{entry.data[CONF_HOST]}:{entry.data[CONF_PORT]}",
+            entry.data[CONF_API_KEY],
+        )
+        self.host = entry.data[CONF_HOST]
+        self.unsub: Callable | None = None
+
+        super().__init__(
+            hass, LOGGER, name=DOMAIN, update_interval=timedelta(seconds=30)
```

Comment:
Do we still need to poll?

Reply:
This handles the reconnect unless there are better options?